### ISSUE TITLE
Lots of buffer breaks 

### ISSUE THREAD

**Issue by Grauen** [yuliskov/SmartTubeNext#658]
I usually use SmartTubeTv and I noticed the video buffering works not so good in SmartTubeNext. 

Most Videos are stuttering with the default setting. Increasing the buffer helps. 

I did not need to adjust any buffer settings in SmartTubeTv to watch videos without any buffer breaks. 

Running on:
Firetv Stick 4k, version 11.28

**Comment by MessiahMozgus** [comment]
It's gotten 10x worse this past couple weeks. We keep going back to Roku and suffer the ads because at least the videos play. The Nvidia shield on cat6 cable to 1GB fiber net should do better.

**Comment by costinmx** [comment]
It's got worse in the past weeks, on Android TV also that's true. Buffering issues are very common lately - some clips, like 1 in 10 videos, cannot buffer properly and is unwatchable even if you set the size to High. It won't go away unless you play another video and then come back again (simple restart won't do). 

Outstanding work regarding Live Stream videos though, thanks to Yuliskov - now they can beautifully play up to 4k.  BTW some of them (very rare) with 4 hours previous buffer are also affected by the buffering issue.

**Comment by finnzz** [comment]
I can confirm I'm seeing the same behavior of constant buffering on FireTV Cube 2nd and the v11.28. it's been getting worse over the last few weeks.  I have buffer set to high, but some videos just don't load/download at the beginning and will buffer continuously.

**Comment by mhh39** [comment]
I also noticed quite a lot of buffering issues lately. It might be google/youtube though, not sure it's an app bug.

Besides that, there's a quite annoying bug when buffering live streams. Apparently, once in a while the live stream is repeating the same content for a few seconds. It's like overlapping the same content, rewinding a couple of seconds and playing it again.
This might be an app bug though. Maybe it's the ads-removal feature that's doing this?

**Comment by finnzz** [comment]
It might be a combination of SmartTubeNext and the YouTube API IDK, but I also use Vanced and I'm not seeing any loading/buffering issues there.  

In SmartTubeNext if a video doesn't load and I go to the previous video and back, the video will load.

**Comment by Kiarasht** [comment]
Happens on Google Chromecast ultra. A single video refuses to load so only way is to just move to another video.

**Comment by f1vefour** [comment]
It happens on my Hisense TV as well, it seems like this didn't happen until sponsor block was integrated.

You can set the buffer to large and it helps a small amount but it still happens a lot.

**Comment by arjenanker** [comment]
Been having similar issues for quite a while now.

First video works fine, but the next one (or sometimes the one after that) will start buffering every couple of seconds. Usually force closing the app and trying again fixes it... for 1 video. Sometimes I have to completely restart my Shield TV for it to start working.

For the record, I see no such issues in the official app or on the webpage. I've tried watching the very same video at the same time in the official app or in a browser with no issues, yet Smarttube constantly buffers. Infuriating.

**Comment by costinmx** [comment]
On Sony Bravia Android TV XG83 (ATV3) everything works fine, as it should, since v11.33 - it looks like one of the best stable updates so far. Not a single buffering problem in 3 days and more than 50 different clips played - regular videos or live stream videos.  Very very rare clip start with "Unknown video decoder error", but it's self correcting and playing starts fine by itself in a couple of seconds. 
I hope this helps the developer to figure out why there may be problems on other platforms. 

As a sugestion: I have noticed that playback decoding and buffering is sometimes highly affected by other apps still open in background. Like Netflix or other players with 4K capabilities.

**Comment by foreveryoungforty** [comment]
Also here everything perfect on Sony Bravia KD-65XF9005 4K

**Comment by galmok** [comment]
Using v11.37 here and I have had to drop to 480p because this app fails to download data fast enough. This has become worse over the last releases. The same device can do 4K/2160p playback on YouTube, but only 480p with this app. I wonder if I have to lower resolution even more for the next release...

**Comment by costinmx** [comment]
It's true. Something happened. "Cant download video". All music videos are not working anymore. Happens on any device

**Comment by stevensmedia** [comment]
It's very frustrating indeed.  Sometimes the app just refuses to buffer at all unless I pause the video, no matter what buffering setting I choose. Still happening as of 11.37.

I can watch a 1080p60 stream on Twitch, but I can't watch a 720p video on SmartTubeNext.

If there's anything I can do to help diagnose this, I'm happy to help.

**Comment by Grauen** [comment]
I just wanted to update my issue - this is still a big issue. Some videos, like this one: https://youtu.be/a4tl8iEAFfk just won't play without breaking playback Evers 3 seconds just to  buffer.

Buffer size is high.
I tried different Codecs in the video profile settings, but it didn't help. 

I hope you find  time to investigate this.

**Comment by Ketec** [comment]
Basically, every video buffers every few seconds. Sometimes stops and restarts playing at the timestamp.
I'm on a 500mbit fiber, TV is on wifi 5 - ~900mbit/s connection (enough to play 4k movies off NAS).
YouTube app itself plays the same content just fine.

There is something broken with buffering or video decoders. It can't seem to even handle 1080p 30 fps content.

**Comment by Muegic** [comment]
Ok thats strange. Updated to  11.37 today and videos played again. But the buffering problem was worse than ever.
So I installed the Vanced Manager on my Shield TV, from there I installed MicroG and Vanced. I enjoyed the quick video loading on Vanced, but the required airmouse drove me crazy. So I opend Smarttubenext again and now videos start playing instant. Not even the „initial 5 sec ad buffering“.
Now Im wondering - did SmarttubeNext fix the buffering problem (for everybody) or is a Vance installation the solution to the buffer nightmare?

**Comment by Daydreamer00001** [comment]
I've resorted to using the old Smart YouTube TV app. At least that works most of the time for me.

**Comment by just4747** [comment]
Exact same problem here. Works fine for several videos and then suddenly videos start buffering every few seconds endlessly. Restarting the video does nothing, neither does changing the video format. I have very fast Internet and a very recent FireTV so those are not issues. SO frustrating. What do we have to do to get this prioritized? It's a complete stopper.

**Comment by f1vefour** [comment]
11.40 seems to have fully fixed both issues, the occasional codec issue and the far worse issue of pausing and/or not starting.

**Comment by just4747** [comment]
> 11.40 seems to have fully fixed both issues, the occasional codec issue and the far worse issue of pausing and/or not starting.

Not at all, not even a little. I've been on it for the past few hours since it got pushed and it's been some of the worst buffering like this that I've ever seen. More frequent than ever and I've ruled out any external causes.

**Comment by MessiahMozgus** [comment]
What was the last known stable version in this regard? This problem is annoying the crap out of my girlfriend, haha. Help.

**Comment by f1vefour** [comment]
> > 11.40 seems to have fully fixed both issues, the occasional codec issue and the far worse issue of pausing and/or not starting.
> 
> Not at all, not even a little. I've been on it for the past few hours since it got pushed and it's been some of the worst buffering like this that I've ever seen. More frequent than ever and I've ruled out any external causes.
> 
> I can see someone falsely thinking it's fixed because it can be intermittent...but when it's happening it's happening bad and non-stop.

You are correct, I still get an initial pause on some 4K videos but worse than that I'm now getting crashes when trying to skip to a different video.

I played some 8 to 10 videos flawlessly and then I started getting pauses and crashes, I was getting crashes on fix 3 version as well.

**Comment by daniel-liuzzi** [comment]
> Some videos, like this one: https://youtu.be/a4tl8iEAFfk just won't play without breaking playback every 3 seconds just to buffer.

Came here to say this video was still unwatchable on 11.40 on a 1Gbps connection over Ethernet; playing 2 seconds and buffering 5, even dropping to 720p, but discovered something weird in the process:

I noticed https://youtu.be/ILHEYblJ7uM played perfectly even at the highest `2160p, 30fps, 15.49Mbit, vp9` profile. I was going to report that this seems to affect only some videos, but then I tried playing https://youtu.be/a4tl8iEAFfk again and this time it played perfectly, even at 4K. It's as if MKBHD's video somehow fixed the buffering issues in the other video.

Like others before me, I also noticed this getting much worse in the past couple weeks. In fact I don't ever remember experiencing this on 11.22.

**Comment by costinmx** [comment]
Sadly, the buffering issue has comed back with a vengeance in 11.40 on my Android, too. Looks like the SmartTube Next had been reverted to an older version, where live streams are once again no longer 4K capable.

**Comment by stevensmedia** [comment]
I've found that when I get buffering, the fix is to completely close the app, re-open it, and then videos play normally.

That's probably a sign of something.

**Comment by costinmx** [comment]
You will get the same result just by playing another video and then play that one again. 
You could press the Previous Track button and then Next Track to have it played again. Basic ideea is that you have to play a different video to unlock buffering issue for that clip.

**Comment by RakeshReddy69** [comment]
Similar issue, when I change accounts random buffer

What worked for me
Close the app >Settings>apps>SmartTubeNext>Clear Cache

Hope this works for you all too and helps the developer fox this issue in the next update

**Comment by RakeshReddy69** [comment]
> You will get the same result just by playing another video and then play that one again.
> You could press the Previous Track button and then Next Track to have it played again. Basic ideea is that you have to play (for a couple of seconds) a different video, then return to unlock buffering issue for that clip.
Thank you...this too works for me..

**Comment by RakeshReddy69** [comment]
> You will get the same result just by playing another video and then play that one again.
> You could press the Previous Track button and then Next Track to have it played again. Basic ideea is that you have to play (for a couple of seconds) a different video, then return to unlock buffering issue for that clip.

Thank you.......this too works for me.....

**Comment by jdotdarko** [comment]
Using AVC codec instead of VP9 also helps. I just noticed this right now. No issues with AVC.

**Comment by just4747** [comment]
> Using AVC codec instead of VP9 also helps. I just noticed this right now. No issues with AVC.

No this doesn't matter. I am on the newest update and neither the update not using AVC format vs the others makes any difference. Even going to the next video and back doesn't always work. As of now, this issue very much still exists no matter what you try, it's just intermittent.

**Comment by RakeshReddy69** [comment]
> > Using AVC codec instead of VP9 also helps. I just noticed this right now. No issues with AVC.
> 
> No this doesn't matter. I am on the newest update and neither the update nor using AVC format vs the others makes any difference. Even going to the next video and back doesn't always work. As of now, this issue very much still exists no matter what you try, it's just intermittent.

Check with 11.42 update...change log mentions fix for buffering issues....this update seems to have fixed the issue..I did experience occasional buffering issue after update to 11.42 but not to the extreme as earlier...also try clearing the apps cache

**Comment by jdotdarko** [comment]
> > Using AVC codec instead of VP9 also helps. I just noticed this right now. No issues with AVC.
> 
> No this doesn't matter. I am on the newest update and neither the update nor using AVC format vs the others makes any difference. Even going to the next video and back doesn't always work. As of now, this issue very much still exists no matter what you try, it's just intermittent.

This does matter because it works for me even with the newest update. Changing from VP9 to AVC while the buffering issue is happening fixed it. I don't know how. I don't know why. But it works on my end.

**Comment by just4747** [comment]
> > > Using AVC codec instead of VP9 also helps. I just noticed this right now. No issues with AVC.
> > 
> > 
> > No this doesn't matter. I am on the newest update and neither the update nor using AVC format vs the others makes any difference. Even going to the next video and back doesn't always work. As of now, this issue very much still exists no matter what you try, it's just intermittent.
> 
> Check with 11.42 update...change log mentions fix for buffering issues....this update seems to have fixed the issue..I did experience occasional buffering issue after update to 11.42 but not to the extreme as earlier...also try clearing the apps cache

As I said I am already on the newest with the "possible buffering fix" and it still happens.

**Comment by just4747** [comment]
> > > Using AVC codec instead of VP9 also helps. I just noticed this right now. No issues with AVC.
> > 
> > 
> > No this doesn't matter. I am on the newest update and neither the update nor using AVC format vs the others makes any difference. Even going to the next video and back doesn't always work. As of now, this issue very much still exists no matter what you try, it's just intermittent.
> 
> This does matter because it works for me even with the newest update. Changing from VP9 to AVC while the buffering issue is happening fixed it. I don't know how. I don't know why. But it works on my end.

Not me man - it's been doing the same abnormal buffering on some videos right away even when it's defaulted to an avc format. Not a fix. Same as before.

**Comment by NGGiovanni** [comment]
> > > Using AVC codec instead of VP9 also helps. I just noticed this right now. No issues with AVC.
> > 
> > 
> > No this doesn't matter. I am on the newest update and neither the update nor using AVC format vs the others makes any difference. Even going to the next video and back doesn't always work. As of now, this issue very much still exists no matter what you try, it's just intermittent.
> 
> Check with 11.42 update...change log mentions fix for buffering issues....this update seems to have fixed the issue..I did experience occasional buffering issue after update to 11.42 but not to the extreme as earlier...also try clearing the apps cache

I'm also having the same issue, it only loads when I restart the device

**Comment by kinghat** [comment]
I haven't had any buffering on Google TV with android on 11.42 and it was previously a constant issue for me for the last couple months.

**Comment by costinmx** [comment]
Also fixed for me too. Sony Android TV XG83. Not a single issue in the past 24hrs, since 11.42

**Comment by f1vefour** [comment]
Using AVC simply makes all videos play in 1080p or lower, VP9 is needed for 
4K and these are the most troublesome videos to play without buffering. I 
have gigabit WAN thus buffering speed is irrelevant.

**Comment by Grauen** [comment]
Problem still appears with 11.42. It's better but still far from acceptable. 

It looks like the developer has just increased the buffer time to 30s but I'm not sure if this is enough...

**Comment by elle8che** [comment]
Still randomly occurring in 11.42, but 11.44 is now out and says "another buffering fix" in the changelog.

**Comment by Grauen** [comment]
I installed 11.45 and since that I had no issues with 1080p videos. I will leave this issue open until I tested 4k videos and if there is nobody else who still has buffer issues.

**Comment by just4747** [comment]
> I installed 11.45 and since then I had no buffer related issues with 1080p videos. I will leave this issue open until I tested 4k videos and if there is nobody else who still has buffer issues.

I wouldn't just assume it's fixed. The issue is very random and intermittent and sometimes you can go a while without it happening and then suddenly it's terrible. Please keep it open, it's still happening to me on the latest version.

**Comment by daniel-liuzzi** [comment]
FWIW, today I tried 11.47 and it _seems_ like buffering issues are fixed. 🤞 

Thanks so much for this awesome project!

**Comment by saschaabraham** [comment]
Not here (firetv 4k stick). Running a 4K HDR Video still does not work. Nerd stats show the start of the buffer fill, only after the buffer is empty. And then is takes only data for 2 seconds and buffers again and again...

**Comment by Grauen** [comment]
> Not here (firetv 4k stick). Running a 4K HDR Video still does not work. Nerd stats show the start of the buffer fill, plays only till the buffer is empty. And then is takes only data for 2 seconds and buffers again and again...

Did you update the buffer size in the Settings?

**Comment by Grauen** [comment]
I had no buffer issues for weeks. I'm closing this issue.

**Comment by Mysticodex** [comment]
Is there a fix @ACR-Jeff ?

**Comment by ACR-Jeff** [comment]
This issue was posted 5 years ago with no current updates on this particular issue, We are in the process of cleaning up the issues, I didn't close this particular issue, The author did. I am organizing and merging duplicates.
